model/work.py

```diff
@@ -781,6 +781,120 @@ def language_code(self):
+
+        # The patron can interact with this Work if any of the audiences
+        # in their root lane (in conjunction with the root lane's target_age)
+        # would match the Work.
+        return any(
+            self.age_appropriate_for(audience, root.target_age)
+            for audience in root.audiences
+        )
+
+    def age_appropriate_for(self, reader_audience=None, reader_age=None):
+        """Is this work age-appropriate for the reader described?
+
+        NOTE: What "age-appropriate" means depends on some policy
+        questions that have not been answered and may be
+        library-specific. For now, non-children's books are
+        age-inappropriate for young children, and children's books are
+        age-inappropriate for children too young to be in the book's
+        target age range.
+
+        :param reader_audience: One of the audience constants from
+           Classifier, representing the general reading audience to
+           which the reader belongs.
+
+        :param reader_age: A number or 2-tuple representing the age or
+           age range of the reader.
+        """
+        if not reader_audience:
+            # A patron with no particular audience restrictions
+            # can see everything.
+            return True
+
+        if reader_audience not in Classifier.AUDIENCES_JUVENILE:
+            # Any non-juvenile patron can see everything.
+            return True
+
+        if self.audience == Classifier.AUDIENCE_ALL_AGES:
+            # An 'all ages' book is always age-appropriate.
+            return True
+
+        # At this point we know that the patron is a juvenile.
+
+        if isinstance(reader_age, tuple):
+            # A range was passed in rather than a specific age. Assume
+            # the reader is at the top edge of the range.
+            ignore, reader_age = reader_age
+
+        # A YA reader is treated as an adult (with no reading
+        # restrictions) if they have no associated age range, or their
+        # age range includes ADULT_AGE_CUTOFF.
+        if (reader_audience == Classifier.AUDIENCE_YOUNG_ADULT
+            and (reader_age is None
+                 or reader_age >= Classifier.ADULT_AGE_CUTOFF)):
+            return True
+
+        # There are no other situations where a juvenile reader can access
+        # non-juvenile titles.
+        if self.audience not in Classifier.AUDIENCES_JUVENILE:
+            return False
+
+        # At this point we know we have a juvenile reader and a
+        # juvenile book.
+
+        if (reader_audience == Classifier.AUDIENCE_YOUNG_ADULT
+            and self.audience in (Classifier.AUDIENCES_YOUNG_CHILDREN)):
+            # A YA reader can see any children's title.
+            return True
+
+        if (reader_audience in (Classifier.AUDIENCES_YOUNG_CHILDREN)
+            and self.audience == Classifier.AUDIENCE_YOUNG_ADULT):
+            # Children cannot see any YA title.
+            return False
```

Comment:
Keep in mind that this method will be called once for every audience in the lane's `.audiences`. Consider the case of 
audiences=[Children, Young Adult], target_age=[9,13], with a YA book with target_age=(12,15).

The first call will be (Children, [9,13]) and it will fail, because children cannot access YA titles even if the age range overlaps. The second call will be (Young Adult, [9,13]) and it will succeed, because a YA reader can access a YA title if the age range overlaps.